clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -1215,6 +1241,69 @@ class ArchiveFileHandler final : public FileHandler {
   Error WriteBundle(raw_fd_ostream &OS, MemoryBuffer &Input) override {
     llvm_unreachable("unsupported for the ArchiveFileHandler");
   }
+
+private:
+  // NOTE: mostly a copy-paste of ReadHeader method.
+  Expected<std::vector<std::string>>
+  ReadTargetsFromChild(const Archive::Child &C) {
+    auto BinOrErr = C.getAsBinary();
+    if (!BinOrErr)
+      return BinOrErr.takeError();
+
+    auto &Bin = BinOrErr.get();
```

Comment:
Note: it is not immediately obvious from the surrounding code, what types those variables have. [Use `auto` if and only if it makes the code more readable or easier to maintain. Don’t “almost always” use `auto`](https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable)